sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -214,12 +221,33 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
     uncacheByCondition(spark, _.sameResult(plan), cascade, blocking)
   }
 
-  def uncacheTableOrView(spark: SparkSession, name: Seq[String], cascade: Boolean): Unit = {
+  private[sql] def uncacheRelation(
```

Comment:
This is used in `UncacheTableExec`. I kept it private as we may migrate that node to use `ResolvedIdentifier` instead of a proper relation. That said, I'd do that in a separate PR.